### PR TITLE
Refactor the python codes part

### PR DESCRIPTION
**Pull request by baotingfang**
We need to refactor the python source codes to get better codes structure. 

**NOTE**

- The `concourse/oss` as our python lib root directory.
- Add the `.gitignore` file
- All python tests files should be placed under `concourse/tests` directory.

This directory structure doesn't the final decision. We can discuss the dir structure in our team and I will change it later.

Details: tracker story: #167354688

Authored-by: Tingfang Bao <bbao@pivotal.io>

### CODE REVIEW

**Comment by baotingfang** [comment]
> Great improvements to code structure!!
> 
> 1. let's move tests/ppa_test.py into oss/ppa folder?

For most python project, a common practice is that put the python unit tests files into a separate directory. eg. tests. It is also good for separating the python business codes from the testing codes.

> 2. why did we choose name for folder as oss/? maybe python/ is better

The package name is not the final decision. We can discuss it in our standown meeting.

**NOTE**

I think the directory structure changes will affect other team member's working job, So I want to merge it first, and if the package name changed, I can fix it later.

**Comment by amilkh** [comment]
+1 to merge it!

File: concourse/pipelines/gpdb_opensource_release.yml [hunk 1]
```diff
@@ -326,7 +326,7 @@ jobs:
       - get: bin_gpdb_ubuntu18.04
         passed: [compile_gpdb_ubuntu18.04]
     - task: push to ppa
-      file: greenplum-database-release/concourse/publish_to_ppa/task.yml
+      file: greenplum-database-release/concourse/tasks/publish_to_ppa.yml
```
Comment (bradfordboyle) [inline concourse/pipelines/gpdb_opensource_release.yml]: Nitpick: I am not a fan of separating task definitions and task scripts into separate directories. I find it hard to navigate from the command line. I think a better concourse convention is:

```sh
concourse/
  task_name/
    task.yaml
    run.{sh, py}
```

I think this convention makes it easier to quickly go from looking at a failure in concourse to the code.

File: concourse/tasks/publish_to_ppa.yml [hunk 2]
```diff
@@ -27,12 +27,14 @@ run:
...
   - |
-    cp greenplum-database-release/concourse/publish_to_ppa/dput.cf ~/.dput.cf
+    cp greenplum-database-release/concourse/scripts/dput.cf ~/.dput.cf
     apt-get update -q=2
     apt-get install -q=2 -y software-properties-common debmake equivs git
     gpg --import <(echo "${GPG_PRIVATE_KEY}")
-    pushd greenplum-database-release/concourse/publish_to_ppa && python3 -m unittest run_test && popd
-    python3 greenplum-database-release/concourse/publish_to_ppa/run.py
+
+    export PYTHONPATH=${PWD}/greenplum-database-release/concourse
```
Comment (bradfordboyle) [inline concourse/tasks/publish_to_ppa.yml]: Do we want to set `PYTHONPATH` or install the package with `pip install -e ./concourse/oss`?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+*.pyc
```
[the remark this answers is not in the document]
Comment (baotingfang) [reply .gitignore]: Yeah. Let's add it.

File: concourse/pipelines/gpdb_opensource_release.yml (hunk 1 above)
Comment (baotingfang) [reply concourse/pipelines/gpdb_opensource_release.yml]: I always thought that the task and scripts directory structure is the best practice for concourse jobs. 

we can move them together in the next commit.

File: concourse/tasks/publish_to_ppa.yml (hunk 2 above)
Comment (baotingfang) [reply concourse/tasks/publish_to_ppa.yml]: Both are ok for me. The target is prepending the `greenplum-database-release/concourse/oss` to the `sys.path`list

File: concourse/pipelines/gpdb_opensource_release.yml (hunk 1 above)
Comment (bradfordboyle) [reply concourse/pipelines/gpdb_opensource_release.yml]: My experience is that it is more of a GPDB convention. I've seen several CloudFoundry teams use `concourse/<task-name>/` and it was always easier to navigate. Just my two cents.

Comment (amilkh) [reply concourse/pipelines/gpdb_opensource_release.yml]: hmm maybe a good pattern to try and bring up in stand-down maybe

**Pull request merged**